keras_nlp/models/roberta/roberta_masked_lm_preprocessor.py

```diff
@@ -0,0 +1,189 @@
+      and a `"</s>"` at the end of the entire sequence.
+    - Randomly select non-special tokens to mask, controlled by
+      `mask_selection_rate`.
+    - Construct a `(x, y, sample_weight)` tuple suitable for training with a
+      `keras_nlp.models.RobertaMaskedLM` task model.
+
+    Args:
+        tokenizer: A `keras_nlp.models.RobertaTokenizer` instance.
+        sequence_length: The length of the packed inputs.
+        mask_selection_rate: The probability an input token will be dynamically
+            masked.
+        mask_selection_length: The maximum number of masked tokens supported
+            by the layer.
+        truncate: string. The algorithm to truncate a list of batched segments
+            to fit within `sequence_length`. The value can be either
+            `round_robin` or `waterfall`:
+                - `"round_robin"`: Available space is assigned one token at a
+                    time in a round-robin fashion to the inputs that still need
+                    some, until the limit is reached.
+                - `"waterfall"`: The allocation of the budget is done using a
+                    "waterfall" algorithm that allocates quota in a
+                    left-to-right manner and fills up the buckets until we run
+                    out of budget. It supports an arbitrary number of segments.
+
+    Examples:
+    ```python
+    # Load the preprocessor from a preset.
+    preprocessor = keras_nlp.models.RobertaMaskedLMPreprocessor.from_preset(
+        "roberta_base_en"
+    )
+
+    # Tokenize and mask a single sentence.
+    sentence = tf.constant("The quick brown fox jumped.")
+    preprocessor(sentence)
+
+    # Tokenize and mask a batch of sentences.
+    sentences = tf.constant(
+        ["The quick brown fox jumped.", "Call me Ishmael."]
+    )
+    preprocessor(sentences)
+
+    # Tokenize and mask a dataset of sentences.
+    features = tf.constant(
+        ["The quick brown fox jumped.", "Call me Ishmael."]
+    )
+    ds = tf.data.Dataset.from_tensor_slices((features))
+    ds = ds.map(preprocessor, num_parallel_calls=tf.data.AUTOTUNE)
+
+    # Alternatively, you can create a preprocessor from your own vocabulary.
+    # The usage is exactly the same as above.
+    vocab = {
+        "<s>": 0, "<pad>": 1, "</s>": 2, "<mask>": 3, "Ġquick": 4, "Ġfox": 5,
+    }
+    merges = ["Ġ q", "u i", "c k", "ui ck", "Ġq uick", "Ġf o", "Ġfo x"]
+    tokenizer = keras_nlp.models.RobertaTokenizer(
+        vocabulary=vocab,
+        merges=merges,
+    )
+    preprocessor = keras_nlp.models.RobertaMaskedLMPreprocessor(
+        tokenizer=tokenizer,
+        sequence_length=20,
+    )
+    preprocessor(" quick fox quick fox")
+    ```
+    """
+
+    def __init__(
+        self,
+        tokenizer,
+        sequence_length=512,
+        mask_selection_rate=0.15,
+        mask_selection_length=96,
+        truncate="round_robin",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
```

Comment:
(Maybe we need this for other preprocessor as well)

Shall we do a type check on the tokenizer? If users pass in a `BertTokenizer`, the code might not error out correctly.

Reply:
Reminds me of #374. Are we the type of library that does a lot of type checking or not? You could imagine adding dozens of type checks to most functions. Is never typechecking cleaner?

Reply:
I could see going either way here. Right now a preprocessor could be a custom layer, which is actually quite useful if you wanted to do some custom masking (whole word masking for example). As long as your inputs and outputs match our prepreprocessor, you are GTG. But the potentially for misuse is higher too for sure!

For now, I will stick with our "prior art" which means no type checking. We can discuss and change this across all task models if we decide to.